src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -277,3 +280,92 @@ def _check_authorization(self, username):
         if not AuthManager.authorize(username, sec_scope, sec_perms):
             raise cherrypy.HTTPError(403, "You don't have permissions to "
                                           "access that resource")
+
+class Oauth2JWTManager:
+    _token = None
+    _token_payload = None
+    _user = None
+
+    @classmethod
+    def initialize_from_token(cls, token):
+        if not JwtManager.is_blocklisted(token):
+            cls._token = token
+            cls._token_payload = cls.get_token_payload()
+            cls._user = cls.get_user()
```

Comment:
When you create a class with only `@classmethod`s and an `init` method that initializes the class variables... Then that's a clear code smell. Objects are usually local, classes are global. Modifying a global class in a multi-threaded context and hoping that it'll behave as a regular object, that's a recipe for disaster.

We either use local objects, or thread-local global objects (like `cherrypy.request`) to store/fetch stuff.

Classes are meant to be instatiated (the JWTManager is an infamous examples and hopefully we'll get rid of it in the near future).

Reply:
Using cherrypy.request now